gen/service_test.go

```diff
@@ -0,0 +1,430 @@
+			isException: keyvalue.DeleteValueHelper.IsException,
+			err:         &tx.DoesNotExistException{Key: "foo"},
+			expected:    true,
+		},
+		{
+			isException: keyvalue.DeleteValueHelper.IsException,
+			err:         errors.New("some error"),
+			expected:    false,
+		},
+	}
+
+	for _, tt := range tests {
+		assert.Equal(t, tt.expected, tt.isException(tt.err))
+	}
+}
+
+func TestWrapResponse(t *testing.T) {
+	tests := []struct {
+		desc           string
+		run            func() (interface{}, error)
+		expectedResult interface{}
+		expectedError  error
+	}{
+		{
+			desc: "setValue success",
+			run: func() (interface{}, error) {
+				return keyvalue.SetValueHelper.WrapResponse(nil)
+			},
+			expectedResult: &keyvalue.SetValueResult{},
+		},
+		{
+			desc: "setValue failure",
+			run: func() (interface{}, error) {
+				return keyvalue.SetValueHelper.WrapResponse(errors.New("foo"))
+			},
+			expectedError: errors.New("foo"),
+		},
+		{
+			desc: "getValue success",
+			run: func() (interface{}, error) {
+				return keyvalue.GetValueHelper.WrapResponse(&tu.ArbitraryValue{BoolValue: boolp(true)}, nil)
+			},
+			expectedResult: &keyvalue.GetValueResult{
+				Success: &tu.ArbitraryValue{BoolValue: boolp(true)},
+			},
+		},
+		{
+			desc: "getValue application error",
+			run: func() (interface{}, error) {
+				return keyvalue.GetValueHelper.WrapResponse(nil, &tx.DoesNotExistException{Key: "foo"})
+			},
+			expectedResult: &keyvalue.GetValueResult{
+				DoesNotExist: &tx.DoesNotExistException{Key: "foo"},
+			},
+		},
+		{
+			desc: "getValue failure",
+			run: func() (interface{}, error) {
+				return keyvalue.GetValueHelper.WrapResponse(nil, errors.New("foo"))
+			},
+			expectedError: errors.New("foo"),
+		},
+		{
+			desc: "deleteValue success",
+			run: func() (interface{}, error) {
+				return keyvalue.DeleteValueHelper.WrapResponse(nil)
+			},
+			expectedResult: &keyvalue.DeleteValueResult{},
+		},
+		{
+			desc: "deleteValue application error (1)",
+			run: func() (interface{}, error) {
+				return keyvalue.DeleteValueHelper.WrapResponse(&tx.DoesNotExistException{Key: "foo"})
+			},
+			expectedResult: &keyvalue.DeleteValueResult{
+				DoesNotExist: &tx.DoesNotExistException{Key: "foo"},
+			},
+		},
+		{
+			desc: "deleteValue application error (2)",
+			run: func() (interface{}, error) {
+				return keyvalue.DeleteValueHelper.WrapResponse(&tv.InternalError{})
+			},
+			expectedResult: &keyvalue.DeleteValueResult{
+				InternalError: &tv.InternalError{},
+			},
+		},
+		{
+			desc: "deleteValue failure",
+			run: func() (interface{}, error) {
+				return keyvalue.DeleteValueHelper.WrapResponse(errors.New("foo"))
+			},
+			expectedError: errors.New("foo"),
+		},
+	}
+
+	for _, tt := range tests {
+		result, err := tt.run()
+		if tt.expectedError != nil {
+			assert.Equal(t, tt.expectedError, err, tt.desc)
+		} else {
+			assert.Equal(t, tt.expectedResult, result, tt.desc)
+		}
+	}
+}
+
+func TestWrapResponseNillError(t *testing.T) {
```

Comment:
nit: typo, s/Nill/Nil. Maybe also make the name more specific, "TestWrapResponseTypedNilError" or something